gamemodes/terrortown/gamemode/client/cl_equip.lua

```diff
@@ -273,26 +272,36 @@ local function CreateEquipmentList(t)
 		local wep = weps[i]
 
 		if wep.IsEquipment and wep:IsEquipment() then
-			owned_ids[#owned_ids + 1] = wep:GetClass()
+			owned_ids[wep:GetClass()] = true
 		end
 	end
 
-	-- Stick to one value for no equipment
-	if #owned_ids == 0 then
-		owned_ids = nil
-	end
-
 	local itms = {}
 	local tmp = GetEquipmentForRole(ply, currole, t.notalive)
 
 	for i = 1, #tmp do
 		if not tmp[i].notBuyable then
+			-- We need them anyways and this way we don't have to calculate them multiple times when sorting
+			tmp[i].cachedEquipmentTranslation = GetEquipmentTranslation(tmp[i].name, tmp[i].PrintName)
 			itms[#itms + 1] = tmp[i]
 		end
 	end
 
 	if #itms == 0 and not t.notalive then
 		ply:ChatPrint("[TTT2][SHOP] You need to run 'shopeditor' as admin in the developer console to create a shop for this role. Link it with another shop or click on the icons to add weapons and items to the shop.")
+	else
+		table.sort(itms, function(itmA, itmB)
```

Comment:
It might be useful to define this function somewhere, so it is easier to find / change and maybe even just use a reference, to be able to swap out sorting functions in the long run.

Reply:
I just found a function that does what mine does in sh_init, but it was only ever used to sort the shopeditor. I dislike the way it sorts though, since it relies solely on the class name. I'll replace it with mine and test if it breaks anything
https://github.com/TTT-2/TTT2/blob/dc859bc6f4efb7a642a6443f337618fb71c3ef93/gamemodes/terrortown/gamemode/shared/sh_init.lua#L416-L438

Reply:
I wanted to emphasize on the possible variable solution to store a sorting function in, so it can be swapped out easily (even programmatcally). I did not want to say you should find an already defined function to replace yours with, just for optimization. You can still keep yours, thats fine, but just define it somewhere in the file instead and use a reference to it (saved on a variable)